xblock/utils/resources.py

```diff
@@ -57,7 +56,8 @@ def render_mako_template(self, template_path, context=None):
         )
         context = context or {}
         template_str = self.load_unicode(template_path)
-        lookup = MakoTemplateLookup(directories=[pkg_resources.resource_filename(self.module_name, '')])
+        directory = os.path.dirname(os.path.realpath(sys.modules[self.module_name].__file__))
```

Comment:
I need to think about this line a bit. It makes me nervous that you are using direct path manipulation instead of the `importlib.resources` API, but I also understand that you probably chose to do that because the API doesn't give any good way to retrieve just the directory name, so path manipulation was the way to go. I'll mull it over and get back to you.

Reply:
Actually I picked earlier line from [here](https://github.com/openedx/XBlock/blob/master/xblock/utils/resources.py#L90)

Update the code though based on importlib package, please review
I think `self.module_name` will always be imported as we are passing it in the `__init__` method of `ResourceLoader`

Reply:
> Update the code though based on importlib package, please review

Nice, that looks a lot better to me.

> I think self.module_name will always be imported as we are passing it in the __init__ method of ResourceLoader

self.module_name is just a string; passing it to `__init__` does not guarantee that the module it points to has been parsed and loaded into `sys.modules` by the import system.